scm/src/GFS_typedefs.F90

```diff
@@ -3783,6 +3762,11 @@ subroutine control_initialize (Model, nlunit, fn_nml, me, master, &
     !--- ak/bk have been flipped from their original FV3 orientation and are defined sfc -> toa
     Model%si = (ak + bk * con_p0 - ak(Model%levr+1)) / (con_p0 - ak(Model%levr+1))
 
+    if (Model%lsm == Model%lsm_noahmp) then
```

Comment:
This is redundant and should be reverted. The same code is executed like 10 lines up.